std/algorithm/searching.d

```diff
@@ -848,7 +848,8 @@ if (isForwardRange!R
     }
 
     //Because of @@@8804@@@: Avoids both "unreachable code" or "no return statement"
-    static if (isInfinite!R) assert(0);
+    static if (isInfinite!R) assert(false, R.stringof ~ "must be an infinite"
```

Comment:
must _not_ be.

Reply:
also just `static assert(!isInfinite!R, msg)`

Reply:
that will no compile, I tried. depending on the type of R that function will not have a return and an assert(false) at the end.

Reply:
Oh, I misread that as a static assert. My bad.